gnocchi/rest/api.py

```diff
@@ -148,18 +148,24 @@ def deserialize(expected_content_types=None):
     return params
 
 
-def validate(schema, data, required=True):
+def validate(schema, data, required=True, detailed_exc=False):
     try:
         return voluptuous.Schema(schema, required=required)(data)
     except voluptuous.Error as e:
-        abort(400, "Invalid input: %s" % e)
+        if detailed_exc:
+            abort(400, {"cause": "Invalid input",
+                        "detail": str(e)})
```

Comment:
… here you can use `e.path` to have details being not a string but a computer understandable path to the key that is wrong.

A string could be constructed for the "reason" field.

Also `cause` has been set usually to "Attribute value error"

Reply:
done